workflows/microshift.py

```diff
@@ -0,0 +1,178 @@
+    req = requests.get(url=url, params={"alt":"media"}, timeout=60)
+    return json.loads(req.content.decode("UTF-8"))
+
+
+def get_job_result(job_run: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Fetches the finished.json and returns a complete dictionary with the job results for dashboard creation.
+    """
+    finished = get_job_finished_json(job_run['path'])
+    return {
+            "num": job_run['num'],
+            "timestamp": finished['timestamp'],
+            "status": finished['result'],
+            "url": f"https://prow.ci.openshift.org/view/gs/test-platform-results/{job_run['path']}"
+        }
+
+
+def get_all_results(job_limit: int) -> Dict[str, List[Dict[str, Any]]]:
+    """
+    Fetches the job results for all versions of MicroShift starting from 4.14 until there are no job runs available for particular version.
+    """
+    logger.info("Fetching job results")
+    fin_results = {}
+    start = time.time()
+
+    # To make the script easier to maintain, we start with oldest version and go up until there are no jobs detected.
+    # That way it won't require an update everytime there's a new release.
+    for minor in range(14, 100):
+        version = f"4.{minor}"
+        runs = get_job_runs_for_version(version, job_limit)
+        logger.info(f"Found {len(runs)} job runs for version {version}")
+
+        if len(runs) == 0:
+            logger.info(f"Assuming that {version} is not being developed yet - stopping collecting the results")
+            break
+
+        results = [get_job_result(run) for run in runs]
+        fin_results[version] = results
+
+    duration = time.time() - start
+    logger.info(f"Took {duration:.2f} seconds to fetch the job results")
+    return dict(sorted(fin_results.items(), reverse=True))
+
+
+def build_microshift_table_rows(version_results: Dict[str, List[Dict[str, Any]]]) -> str:
+    output = ""
+    for version, results in version_results.items():
+        output += build_microshift_table_row(version, results)
+    return output
+
+
+def build_microshift_table_row(version: str, results: List[Dict[str, Any]]) -> str:
+    """
+    Build a small HTML snippet that displays info about GPU bundle statuses
+    (shown in a 'history-bar' with colored squares).
+    """
+    if len(results) == 0:
+        return ""
+
+    sorted_results = sorted(results, key=lambda r: r["timestamp"], reverse=True)
+    latest_result = sorted_results[0]
+    latest_result_date = datetime.datetime.fromtimestamp(int(latest_result["timestamp"]), datetime.timezone.utc).strftime("%Y-%m-%d %H:%M:%S UTC")
+
+    output = f"""
+        <tr>
+          <td style="min-width:150px; white-space:nowrap;">MicroShift {version}</td>
+          <td>
+            <div class="history-bar" style="border: none; margin: 0; padding: 0; background-color: transparent; box-shadow: none;">
+              <div style="margin-top: 5px;">
+                <strong>Latest run:</strong> {latest_result_date}
+              </div>
```

Comment:
Can we reuse the styles from [styles.css](https://github.com/rh-ecosystem-edge/nvidia-ci/blob/ec63f55ab0b3b409ca3dc0e3c52e9b348dd6f630/styles.css) here? Or maybe add a new style if what you need is missing? I see it's the same in the existing dashboard, so maybe there's an opportunity for improvement :)

Reply:
PTAL: https://github.com/rh-ecosystem-edge/nvidia-ci/pull/234/commits/037dfc7bae718463ffa8678286714cfaf0fd0670

However it requires following change in the styles.css. Any other way to update it beside PR against `gh-pages` branch?
```css
.version-cell {
    min-width: 150px;
    white-space: nowrap;
}

/* History bar styling for GPU bundle (de-emphasized) */
.history-bar-outer {
    margin: 20px auto;
    padding: 10px;
    border: 1px solid #ccc;
    border-radius: 4px;
    background-color: #f7f7f7;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05);
    opacity: 0.7;
}

.history-bar-inner {
    display: flex;
    align-items: center;
    gap: 10px;
    border-radius: 4px;
    color: #888;
    font-size: 0.9em;
    flex-wrap: wrap;
}
```